### ISSUE TITLE
Client callback not fired when provide async function

### ISSUE THREAD

**Issue by koteesy** [grpc/grpc-node#1763]
### Problem description
Client callback and event on('data') not fired when provide async function callback to client method

### Reproduction steps
For example, we have that call:
```
trackingService.parse(request, (err, response) => {
  console.log('Got response!');
});
```

This will print in our console ``` Got response! ```

But if we add async declaration:
```
trackingService.parse(request, async (err, response) => {
  console.log('Got response!');
});
```

This cause infinite loading, and callback will never prints.

### Environment
 - OS name: OSx 11.2.3 (Macbook air)
 - Node version: v14.16.0
 - Package name and version ["grpc": "^1.24.7"]

**Comment by murgatroid99** [comment]
The `grpc` package has been deprecated. I don't know what causes this problem, and at this point I don't think we're going to put in work to fix it. We recommend using `@grpc/grpc-js` instead. Do you experience the same problem with that library?

**Comment by murgatroid99** [comment]
It looks like the fix for this bug in `grpc` is in #1786, and the corresponding fix for `@grpc/grpc-js` is in #1787, which has been published in version 1.3.2.
